### PR TITLE
Add lineprotocol format to influx CLI.

### PR DESCRIPTION
**Pull request by martin-magakian**
###### Required for all non-trivial PRs
- [x] Rebased/mergable
- [x] Tests pass
- [x] CHANGELOG.md updated
- [x] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)

Add lineprotocol format to influx CLI.

### CODE REVIEW

File: cmd/influx/cli/cli.go [hunk 1]
```diff
@@ -642,6 +644,114 @@ func (c *CommandLine) writeColumns(response *client.Response, w io.Writer) {
...
+
+		for _, multiRow := range result.Series {
+			for _, row := range c.formatLineProtocol(multiRow) {
+				fmt.Fprintln(w, row)
+			}
+		}
+	}
+}
+
+func (c *CommandLine) formatLineProtocol(row models.Row) []string {
```
Comment (phemmer) [inline cmd/influx/cli/cli.go]: Did you look into using the existing serializer ([`"github.com/influxdata/influxdb".Point.String()`](github.com/influxdata/influxdb))? Would make the code much smaller, and be better to not have multiple serializers for the same format.

File: influxdb.iml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<module version="4">
+  <component name="NewModuleRootManager" inherit-compiler-output="false">
+    <orderEntry type="sourceFolder" forTests="false" />
+  </component>
+</module>
```
Comment (phemmer) [inline influxdb.iml]: What is this?

Comment (martin-magakian) [inline influxdb.iml]: my bad, I will remove it. (intellij file)

**Comment by martin-magakian** [comment]
Thanks @phemmer to point me in the right direction.

Can anyone help with the continuous-integration?
It look like something went wrong...

**Comment by martin-magakian** [comment]
I don't know what's happen with "continuous-integration/appveyor/pr" but it look ok after a rebase.

**Comment by rhysrhaven** [comment]
so i started using this by building your branch.

problem is that the line protocol you're outputting seems incorrect. You're smashing tags into fields.
the correct format: 

`<measurement>[,<tag_key>=<tag_value>[,<tag_key>=<tag_value>]] <field_key>=<field_value>[,<field_key>=<field_value>] [<timestamp>]`

here is a line from the output of 
`./influx -database telegraf -execute 'SELECT * FROM "cpu" ORDER BY time DESC' -format lineprotocol`

`cpu cluster="sym3",cpu="cpu0",environment="dev",host="i-redacted",ipv4="redacted",island="dev",role="redacted",service=,usage_guest="0",usage_guest_nice="0",usage_idle="88.67346938519503",usage_iowait="0.10204081632100016",usage_irq="1.1224489794753183",usage_nice="0",usage_softirq="3.163265305858199",usage_steal="0.20408163263271975",usage_system="3.6734693873332724",usage_user="3.0612244894443936" 1474318280008424563`

File: cmd/influx/cli/cli.go [hunk 3]
```diff
@@ -642,6 +645,51 @@ func (c *CommandLine) writeColumns(response *client.Response, w io.Writer) {
...
+			points := c.rowToPoints(row)
+			for _, point := range points {
+				fmt.Fprintln(w, point.String())
+			}
+		}
+	}
+}
+
+func (c *CommandLine) rowToPoints(row models.Row) models.Points {
+	tags := models.NewTags(row.Tags)
```
Comment (rhysrhaven) [inline cmd/influx/cli/cli.go]: testing here, `row.Tags` is empty, and all tags are included `row.Columns`

**Comment by rhysrhaven** [comment]
and the problem here makes sense because the underlying API does not distinguish tags vs fields. 
```curl localhost:8086/query?db=dcos --data-urlencode 'q=select * from "cpu" limit 1'
{"results":[{"series":[{"name":"cpu","columns":["time","cluster","cpu","environment","host","ipv4","island","role","service","usage_guest","usage_guest_nice","usage_idle","usage_iowait","usage_irq","usage_nice","usage_softirq","usage_steal","usage_system","usage_user"],"values":[["2016-07-14T10:05:40.001823457Z","sym3","cpu0","dev","i-redacted","redacted","dev","redacted",null,0,0,86.55967903483926,0,0.4012036108051488,0,0.30090270810158104,0,6.920762286418465,5.817452356697464]]}]}]}```

**Comment by martin-magakian** [comment]
Thanks @rhysrhaven for your review.
Should probably request the tags before providing the CLI output.

> show tag keys from "cpu"

**Comment by rhysrhaven** [comment]
That would make the code rather complex imo. You dont know what the query is going to return, and having to make separate calls to figure out which are tags for which line protocol outputs is going to be ugly.

Most likely the API will have to be modified to support a mode where it returns tags and fields separately. 

For my own usage, I found this tool to dump TSM files to lineprotocol. https://docs.influxdata.com/influxdb/v1.0/tools/influx_inspect/

**Comment by corylanou** [comment]
I think it would be better to change the response writer in the server, so that you could use `curl` and other tools to get line protocol back, and not limit it to the CLI tool.

See here for how we implemented csv/column formats: https://github.com/influxdata/influxdb/blob/master/services/httpd/response_writer.go

**Comment by corylanou** [comment]
Per my previous comment, this should be implemented in the response writer in influxd.  Please re-open a new pull request with that implementation.  Thanks for your hard work and contribution.

**Pull request closed without merging**